src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -280,29 +339,36 @@ private static void httpRequestHandler(RelayedHttpListenerContext context, Strin
 	}
 	
 	private static void httpRequestSender(String method, String msgExpected, String msgToSend) throws IOException {
+		HttpURLConnection connection = getHttpConnection();
+		sendBytesOverHttp(connection, msgToSend.getBytes(), 0, msgToSend.length());
+		
+		String received = new String(readBytesFromStream(connection.getInputStream()));
+		assertEquals("Http connection sender did not receive the expected response code.", STATUS_CODE, connection.getResponseCode());
+		assertEquals("Http connection sender did not receive the expected response description.", STATUS_DESCRIPTION, connection.getResponseMessage());
+		assertEquals("Http connection sender did not receive the expected response message size.", 
+				msgExpected.length(), 
+				received.length());
+		assertEquals("Http connection sender did not receive the expected response message.", msgExpected, received);
+	}
+	
+	private static HttpURLConnection getHttpConnection() throws IOException {
 		StringBuilder urlBuilder = new StringBuilder(TestUtil.RELAY_NAMESPACE_URI + TestUtil.ENTITY_PATH);
 		urlBuilder.replace(0, 5, "https://");
 		URL url = new URL(urlBuilder.toString());
 		String tokenString = tokenProvider.getTokenAsync(url.toString(), Duration.ofHours(1)).join().getToken();
 		
-		HttpURLConnection conn = (HttpURLConnection)url.openConnection();
-		conn.setRequestMethod(method);
-		conn.setRequestProperty("ServiceBusAuthorization", tokenString);
-		
-		String message = msgToSend;
-		conn.setDoOutput(true);
-		OutputStreamWriter out = new OutputStreamWriter(conn.getOutputStream());
-		out.write(message, 0, message.length());
+		HttpURLConnection connection = (HttpURLConnection)url.openConnection();
+		connection.setRequestProperty("ServiceBusAuthorization", tokenString);
+		return connection;
+	}
+	
+	private static void sendBytesOverHttp(HttpURLConnection connection, byte[] b, int off, int len) throws IOException {
+		connection.setRequestMethod((len > 0) ? "POST" : "GET");
```

Comment:
>( [](start = 30, length = 1)

total style nit but consider omitting these parenthesis if they're not needed.

Reply:
I said this then I started to feel a bit silly. (At this point you have more Java experience on the team than I do and perhaps the Java conventions should come from you). Feel free to ignore. We do have a .NET StyleCop rule which enforces this in C# though. 

---
In reply to: [294385721](https://github.com/Azure/azure-relay-java/pull/47#discussion_r294385721) [](ancestors = 294385721)